web/src/js/__tests__/ducks/modes/regularSpec.tsx

```diff
@@ -0,0 +1,123 @@
+import regularReducer, {
+    toggleRegular,
+    getMode,
+    initialState,
+    MODE_REGULAR_TOGGLE,
+} from "./../../../ducks/modes/regular";
+import * as options from "../../../ducks/options";
+import { TStore } from "../tutils";
+import fetchMock, { enableFetchMocks } from "jest-fetch-mock";
+
+describe("regularReducer", () => {
+    it("should return the initial state", () => {
+        const state = regularReducer(undefined, {});
+        expect(state).toEqual(initialState);
+    });
+
+    it("should dispatch MODE_REGULAR_TOGGLE and updateMode", async () => {
+        enableFetchMocks();
+
+        const store = TStore();
+
+        expect(store.getState().modes.regular.active).toBe(true);
+        await store.dispatch(toggleRegular());
+        expect(store.getState().modes.regular.active).toBe(false);
+        expect(fetchMock).toHaveBeenCalled();
+    });
+
+    it('should handle RECEIVE_OPTIONS action with data.mode containing "regular", an host and a port', () => {
+        const action = {
+            type: options.RECEIVE,
+            data: {
+                mode: {
+                    value: ["regular@http:8081"],
```

Comment:
Let's use realistic values - `http` does not conceptually make sense as listen_host. We can just go with `127.0.0.1`. It doesn't matter for the test, but inevitably some new contributor will grep through the codebase and be confused by this.

Reply:
yes typo, I use `localhost` which is used below as well :)